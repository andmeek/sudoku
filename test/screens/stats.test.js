import Stats from '../../src/screens/stats.vue'
import { shallowMount } from '@vue/test-utils'

describe('Stats.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Stats)
  })

  test('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('emits an exit event when the user clicks back', () => {
    wrapper.find({ ref: 'exit' }).trigger('click')

    expect(wrapper.emitted().exit).toBeTruthy()
  })
})
