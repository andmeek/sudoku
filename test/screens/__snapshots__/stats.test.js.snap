// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats.vue matches snapshot 1`] = `
<div class="stats screen">
  <header>
    <h1>Gameplay Stats</h1> <button class="menu">
      Back
    </button>
  </header>
  <table>
    <tr>
      <th>Difficulty</th>
      <th>Plays</th>
      <th>Completed</th>
      <th>Best Time</th>
    </tr>
    <tr>
      <td>Easy</td>
      <td></td>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <td>Medium</td>
      <td></td>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <td>Hard</td>
      <td></td>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <td>Very Hard</td>
      <td></td>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <td>Insane</td>
      <td></td>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <td>Unthinkable</td>
      <td></td>
      <td></td>
      <td></td>
    </tr>
  </table>
  <h2>Plays</h2>
  <ul></ul> <button class="menu">
    Wipe Stats
  </button>
</div>
`;

exports[`Stats.vue matches snapshot with game play history 1`] = `
<div class="stats screen">
  <header>
    <h1>Gameplay Stats</h1> <button class="menu">
      Back
    </button>
  </header>
  <table>
    <tr>
      <th>Difficulty</th>
      <th>Plays</th>
      <th>Completed</th>
      <th>Best Time</th>
    </tr>
    <tr>
      <td>Easy</td>
      <td></td>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <td>Medium</td>
      <td></td>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <td>Hard</td>
      <td></td>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <td>Very Hard</td>
      <td></td>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <td>Insane</td>
      <td></td>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <td>Unthinkable</td>
      <td></td>
      <td></td>
      <td></td>
    </tr>
  </table>
  <h2>Plays</h2>
  <ul>
    <li>
      2019-9-5 - Hard, completed in 34193:54:58, with 0 mistakes.
    </li>
    <li>
      1969-12-31 - Easy, incompleted in 00:00:20, with 2 mistakes.
    </li>
  </ul> <button class="menu">
    Wipe Stats
  </button>
</div>
`;

exports[`Stats.vue matches snapshot with game plays stats 1`] = `
<div class="stats screen">
  <header>
    <h1>Gameplay Stats</h1> <button class="menu">
      Back
    </button>
  </header>
  <table>
    <tr>
      <th>Difficulty</th>
      <th>Plays</th>
      <th>Completed</th>
      <th>Best Time</th>
    </tr>
    <tr>
      <td>Easy</td>
      <td>20</td>
      <td>2</td>
      <td>00:02:03</td>
    </tr>
    <tr>
      <td>Medium</td>
      <td></td>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <td>Hard</td>
      <td>2</td>
      <td>1</td>
      <td>34:12:02</td>
    </tr>
    <tr>
      <td>Very Hard</td>
      <td>1</td>
      <td>0</td>
      <td></td>
    </tr>
    <tr>
      <td>Insane</td>
      <td></td>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <td>Unthinkable</td>
      <td></td>
      <td></td>
      <td></td>
    </tr>
  </table>
  <h2>Plays</h2>
  <ul></ul> <button class="menu">
    Wipe Stats
  </button>
</div>
`;
